controllers/imagerepository_controller.go

```diff
@@ -230,6 +234,71 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return ctrl.Result{}, nil
 }
 
+func (r *ImageRepositoryReconciler) HandleNotifications(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) error {
+	log := ctrllog.FromContext(ctx).WithName("HandleNotifications")
+
+	if imageRepository.Status.Notifications == nil {
+		// No status notifications to check
+		return nil
+	}
+
+	log.Info("Handling notifications")
+	allNotifications, err := r.QuayClient.GetNotifications(r.QuayOrganization, imageRepository.Spec.Image.Name)
+	if err != nil {
+		return err
+	}
+	for index, statusNotification := range imageRepository.Status.Notifications {
```

Comment:
This section seems to deserve some refactoring to avoid so many nested layers of `for` loops and `if` statements. :D

Reply:
I couldn't find any other way to do all the comparisons between Spec.Notifications, Status.Notifications and Quay notifications :thinking: